esphome/components/growatt_solar/sensor.py

```diff
@@ -163,6 +165,9 @@ async def to_code(config):
 
     cg.add(var.set_protocol_version(config[CONF_PROTOCOL_VERSION]))
 
+    if CONF_WARM_UP_TIME in config:
+        cg.add(var.set_warm_up_time(int(config[CONF_WARM_UP_TIME])))
```

Comment:
There is a default of 0 so it is always there
```suggestion
    cg.add(var.set_warm_up_time(int(config[CONF_WARM_UP_TIME])))
```

Reply:
ah okay I misunderstood it then, I thought `config` would only be the config text blob without the defaults.